pkg/transform/k8s.go

```diff
@@ -103,10 +115,37 @@ func appendMetadata(span *request.Span, info *kube.PodInfo) {
 		attr.K8sNodeName:      info.NodeName,
 		attr.K8sPodUID:        string(info.UID),
 		attr.K8sPodStartTime:  info.StartTimeStr,
+		attr.K8sClusterName:   KubeClusterName(md.ctx, md.cfg),
```

Comment:
Executing this function on each span decoration would slowdown a lot Beyla. I'd do:

1. Add `clusterName string` as attribute of `metadataDecorator` struct and use the value directly here.
2. In the `KubeDecoratorProvider` function, invoke `KubeClusterName` function once and store the result in the `clusterName` field.